generator/src/main/java/org/stjs/generator/javac/ElementUtils.java

```diff
@@ -445,4 +445,23 @@ public static boolean hasMultipleConstructors(Element element) {
 		return constructorCount > 1;
 	}
 
+	public static boolean isClass(Element element) {
+		return (element != null) && (element.getKind() == ElementKind.CLASS);
+	}
+
+	public static boolean isInnerClass(Element element) {
+		return isClass(element) && isClass(element.getEnclosingElement());
```

Comment:
Checking for a "," in the name is a bad shortcut and was not working for nested inner classes. Checking for the enclosing element is safer and works in all situations.